tests/test_tree_features.py

```diff
@@ -203,6 +203,60 @@ def test_min_samples_split_setting():
         assert min_samples_split_desired <= node.parent.n_samples, f"Failed as node had a parent with {min_samples_split_desired}, but which should have been a lead node"
 
 
+def test_max_features_errors():
+    max_feature_vals = [-1, "cos", []]
```

Comment:
I have already updated this on the Dev branch from the issues I had here.
Going to try and run a merge from Dev to this branch to fix the issue.